packages/snaps-controllers/src/interface/utils.ts

```diff
@@ -111,12 +114,22 @@ function getComponentStateValue(
     | DropdownElement
     | RadioGroupElement
     | CheckboxElement
-    | SelectorElement,
+    | SelectorElement
+    | AddressInputElement,
 ) {
   switch (element.type) {
     case 'Checkbox':
       return element.props.checked;
 
+    case 'AddressInput': {
+      if (!element.props.value) {
+        return null;
+      }
+
+      // Construct CAIP-10 Id
+      const { namespace, reference } = parseCaipChainId(element.props.chainId);
+      return toCaipAccountId(namespace, reference, element.props.value);
```

Comment:
Since we already know that the `chainId` is valid from struct validation, wouldn't it be easier to just do:
```suggestion
      return `${element.props.chainId}:${element.props.value}`;
```

Reply:
Oh I see @GuillaumeRx asked for this, care to elaborate?

Reply:
I just prefer to use the utils we have across the board, makes things easier to understand

Reply:
Yeah, that makes sense. This double validates the chain ID part though, so I am a little worried about that, but we can always optimize if need be.